pyiron_contrib/atomistics/atomicrex/structure_list.py

```diff
@@ -26,33 +26,49 @@ class FlattenedStructureContainer:
     Structures are stored together with an identifier that should be unique.
     The class has to be initialized with the number of structures and the total number of atoms in all structures.
     This allows to preallocate the necessary arrays and therefore prevents reallocations.
-
-    TODO: Alternatively preallocate arrays with large default values (f.e. 1e6 atoms) then use resize to shrink them?
-    """    
+    """
     def __init__(self, num_structures=0, num_atoms=0):
-        self.num_structures = num_structures
-        self.num_atoms = num_atoms
+        self._num_structures_alloc = num_structures
+        self._num_atoms_alloc = num_atoms
         # store the starting index for properties with unknown length
         self.current_atom_index = 0
         # store the index for properties of known size, stored at the same index as the structure
         self.current_structure_index = 0
         # Also store indices of structure recently added
         self.prev_structure_index = 0
         self.prev_atom_index = 0
-        
+
         self._init_arrays()
 
 
     def _init_arrays(self):
-        self.symbols = np.full(self.num_atoms, "XX", dtype=np.dtype("U2"))# 2 character unicode array for chemical symbols
-        self.positions = np.empty((self.num_atoms, 3))
-        self.cells = np.empty((self.num_structures, 3, 3))
-        self.start_indices = np.empty(self.num_structures, dtype=np.int32)
-        self.len_current_struct = np.empty(self.num_structures, dtype=np.int32)
-        self.identifiers = np.empty(self.num_atoms, dtype=np.dtype("U20"))
-
+        self.symbols = np.full(self._num_atoms_alloc, "XX", dtype=np.dtype("U2"))# 2 character unicode array for chemical symbols
+        self.positions = np.empty((self._num_atoms_alloc, 3))
+        self.cells = np.empty((self.__num_structures_alloc_alloc, 3, 3))
+        self.start_indices = np.empty(self.__num_structures_alloc_alloc, dtype=np.int32)
+        self.len_current_struct = np.empty(self.__num_structures_alloc_alloc, dtype=np.int32)
+        self.identifiers = np.empty(self.__num_structures_alloc_alloc, dtype=np.dtype("U20"))
```

Comment:
I still would like to keep num_structures, but if we decide otherwise it should be _num_structures_alloc and not __num_structures_alloc_alloc I guess.

Reply:
:S